pkg/ai/iai.go

```diff
@@ -1,8 +1,20 @@
 package ai
 
-import "context"
+import (
+	"context"
+)
 
 type IAI interface {
 	Configure(token string, language string) error
 	GetCompletion(ctx context.Context, prompt string) (string, error)
+	Parse(ctx context.Context, prompt []string, nocache bool) (string, error)
+}
+
+func NewClient(provider string) IAI {
```

Comment:
Should the client's choice be located at the interface level? Shouldn't we move it to the client file instead?

Reply:
As it is decided here which client will be created, I think it's good to have it here ... But to be honest, I'm still looking for a way to make this cleaner :-)

Reply:
Personally, I prefer having my client declaration at the same level as my methods (similar to how Vault does it with its [client](https://github.com/hashicorp/vault/blob/main/api/client.go#L584)). But it's a subjective choice that doesn't block the validation of this code 😄